ha/core/support_bundle/ha_support_bundle.py

```diff
@@ -0,0 +1,156 @@
+            f'cortx{_DELIM}common{_DELIM}storage{_DELIM}local')
+        machine_id = Conf.machine_id
+
+        if os.path.exists(_sb_tmp_src):
+            HASupportBundle.__clear_tmp_files()
+        else:
+            os.makedirs(_sb_tmp_src)
+        # Copy log files
+        shutil.copytree(os.path.join(log_base, f'ha/{machine_id}'),\
+            os.path.join(_sb_tmp_src, 'logs'))
+
+        # Copy configuration files
+        shutil.copytree(os.path.join(local_base, 'ha'),\
+            os.path.join(_sb_tmp_src, 'conf'))
+        HASupportBundle.__generate_tar(bundle_id, target_path)
+        HASupportBundle.__clear_tmp_files()
+
+    @staticmethod
+    def __copy_file(source: str, destination: str = None):
+        """Copy a file from source to destination location."""
+        directory = os.path.dirname(_sb_tmp_src)
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+        if destination is None:
+            destination = os.path.join(_sb_tmp_src,
+                os.path.basename(source))
+        try:
+            shutil.copy2(source, destination)
+        except FileNotFoundError as fe:
+            raise SupportBundleError(f"File not found : {fe}")
+
+    @staticmethod
+    def __generate_tar(bundle_id: str, target_path: str):
+        """Generate tar.gz file at given path."""
+        component = 'ha'
+        target_path = target_path if target_path is not None \
+            else _sb_default_path
+        target_path = os.path.join(target_path, component)
+        tar_name = bundle_id if bundle_id else _sb_tar_name
+        tar_file_name = os.path.join(target_path, tar_name + '.tar.gz')
+        if not os.path.exists(target_path):
+            os.makedirs(target_path)
+        with tarfile.open(tar_file_name, 'w:gz') as tar:
+            tar.add(_sb_tmp_src,
+                arcname=os.path.basename(_sb_tmp_src))
+
+    @staticmethod
+    def __clear_tmp_files():
+        """Clean temporary files created by the support bundle."""
+        shutil.rmtree(_sb_tmp_src)
+
+    @staticmethod
+    def parse_args():
+        """Parsing support bundle arguments."""
+        parser = argparse.ArgumentParser(description='''Bundle ha logs ''')
+        parser.add_argument('-b', dest='bundle_id', help='Unique bundle id')
+        parser.add_argument('-t', dest='path', help='Path to store the created bundle',
+            nargs='?', default="/var/seagate/cortx/support_bundle/")
+        parser.add_argument('-c', dest='cluster_conf',\
+            help="Cluster config file path for Support Bundle")
+        parser.add_argument('-s', '--services', dest='services', nargs='+',\
+            default='', help='List of services for Support Bundle')
+        parser.add_argument('-d', '--duration', default='P5D', dest='duration',
+            help="Duration - duration for which log should be captured, Default - P5D")
+        parser.add_argument('--size_limit', default='500MB', dest='size_limit',
```

Comment:
How are these default values derived ? Is there any guideline for these numbers ?

Reply:
I have preferred https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/749076684/Node+level+Support+Bundle+Collection confluence page.